fleet_adapter_r3/fleet_adapter_r3/RobotClientAPI.py

```diff
@@ -531,34 +534,21 @@ def navigate_robot(self, robot_encoding_id: str, time_stamp: float, content: Nav
 
     def navigation_completed(self, robot_name: str) -> NavigationStatus:
         self.refresh_expired_token()
-        
-        with self._lock:
-            mission_status = self.get_mission_status(robot_name=robot_name)
-            robot_status = self.get_robot_status(robot_name=robot_name)
 
-        if mission_status is None or robot_status is None:
-            return NavigationStatus.EMPTY
-        
-        robot_mission_status = mission_status['missionStatus']['mission']['status']
-
-        if robot_mission_status == RobotMissionStatus.MOVING_FINISHED.value:
-            # 5 seconds to wait for any error code
-            # This is necessary as mission status and response codes are from two separate message payloads
-            retries = 5
-            while retries > 0:
-                # within this time window, if any error code comes in, navigation was not completed
-                if ResponseCode.P2P_STARTED not in mission_status['alertIds']:
-                    return NavigationStatus.NAVIGATION_ERROR
-                
-                time.sleep(1)
-                retries -= 1
+        threshold = self.waypoint_buffer_threshold
 
-            # P2P_STARTED indicates moving started and finished successfully without any errors in between
-            return NavigationStatus.NAVIGATION_SUCCESS
-                
+        curr_robot_position = self.position(robot_name)
```

Comment:
Would prefer to use the robot position based on the websocket payload rather than calling the position api